### PR TITLE
Spring Boot Web Task 2 done.

### PR DESCRIPTION
**Pull request by yurlis**
But HINT recommendation i didn't use. Use pom.xml configuration from lessons video.

### CODE REVIEW

File: src/main/java/com/bookstoreapp/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "api/books")
```
Comment (koval-kn) [inline src/main/java/com/bookstoreapp/controller/BookController.java]: extract `api/` to the application properties `server.servlet.context-path=/api` or somehow property in order not to duplicate it each time in each controller
```suggestion
@RequestMapping("/books")
```

File: src/main/java/com/bookstoreapp/model/Book.java [hunk 2]
```diff
@@ -9,8 +9,8 @@
 import java.math.BigDecimal;
 import lombok.Data;
 
-@Entity
 @Data
```
Comment (koval-kn) [inline src/main/java/com/bookstoreapp/model/Book.java]: dont use data on entites as it cause bugs, use getter/setter instead

File: src/test/java/com/bookstoreapp/BookstoreApplicationTests.java [hunk 3]
```diff
@@ -5,9 +5,7 @@
...
 @SpringBootTest
 class BookstoreApplicationTests {
-
-	@Test
-	void contextLoads() {
-	}
-
+    @Test
+    void contextLoads() {
+    }
```
Comment (koval-kn) [inline src/test/java/com/bookstoreapp/BookstoreApplicationTests.java]: redundant test, better remove it

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: src/main/java/com/bookstoreapp/repository/BookDaoImpl.java [hunk 4]
```diff
@@ -1,47 +1,48 @@
...
     }
 
     @Override
     public List<Book> findAll() {
-        try (Session session = sessionFactory.openSession()) {
-            return session.createQuery(
-                    "SELECT b FROM Book b", Book.class).getResultList();
-        } catch (Exception e) {
-            throw new RuntimeException("Can't get all users from DB", e);
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
```
Comment (MAMentorFX) [inline src/main/java/com/bookstoreapp/repository/BookDaoImpl.java:36]: Should it have catch block?

**Review by MAMentorFX — APPROVED** [review]

**Pull request merged**